_posts/2024-10-15-Announcing-Data-Prepper-2.10.0.md

```diff
@@ -0,0 +1,129 @@
+A new source has been added in Data Prepper named `opensearch_api` that accepts [OpenSearch Document API Bulk operation](https://opensearch.org/docs/latest/api-reference/document-apis/bulk/) requests from clients using REST and ingests data into OpenSearch. 
+The behavior of this source is also quite similar to the existing `http` source. 
+It supports industry-standard encryption in the form of TLS/HTTPS and HTTP basic authentication. 
+It parses incoming requests and create Data Prepper events and associated event metadata making it compatible with the `opensearch` sink. 
+The request body should be compatible with OpenSearch Document API Bulk Operation and will also support all the actions like index, create, delete and update.
+
+The two HTTP methods supported now are the following:
+
+```
+POST _bulk
+POST <index>/_bulk
+```
+
+The second API which specifies the index in the path means you don’t need to include it in the request body.
+
+Moreover, the following query parameters are also supported that are available in OpenSearch Document API Bulk Operation as the following below:
+
+* pipeline
+* routing
+
+An example of using the source
+
+```
+version: "2"
+opensearch-api-pipeline:
+  source:
+    opensearch_api:
+  sink:
+    - opensearch:
+        hosts: ["https://opensearch:9200"]
+        insecure: true
+        username: "admin"
+        password: "admin"
+        index: "${getMetadata(\"opensearch_index\")}"
+        action: "${getMetadata(\"opensearch_action\")}"
+        document_id: "${getMetadata(\"opensearch_id\")}"
+        routing: "${getMetadata(\"opensearch_routing\")}"
+        pipeline: "${getMetadata(\"opensearch_pipeline\")}"
+```
+
+Consider the sample:
+
+```
+POST _bulk
+{ "index": { "_index": "movies", "_id": "tt1979320" } }
+{ "title": "Rush", "year": 2013 }
+```
+
+The above request will be ingested into OpenSearch and a new document will be created under the index movies with a document id `tt1979320` with the document `{ "title": "Rush", "year": 2013 }`.
+
+Over time, the Data Prepper maintainers are interested in expanding this source to support other indexing APIs to allow it to stand-in for an OpenSearch cluster for ingestion workloads.
+To learn more or express interest see [Provide an OpenSearch API source #4180](https://github.com/opensearch-project/data-prepper/issues/4180).
+
+
+## Kinesis source
+
+[Amazon Kinesis Data Streams](https://docs.aws.amazon.com/streams/latest/dev/introduction.html) is a high speed streaming data service. 
+Data Prepper is introducing a new source named `kinesis` which can be used to ingest stream records data from multiple Kinesis data streams into OpenSearch clusters. 
+You can configure it to read stream records from the beginning or from the latest record. 
+Moreover, if you enable end to end acknowledgements, Kinesis data streams will be checkpointed to prevent duplicate processing of records.
```

Comment:
```suggestion
Moreover, if you enable end-to-end acknowledgements, Kinesis data streams will be checkpointed to prevent duplicate processing of records.
```